lib/engines/content_block_manager/app/services/content_block_manager/get_preview_content.rb

```diff
@@ -0,0 +1,40 @@
+require "net/http"
+require "json"
+require "uri"
+
+module ContentBlockManager
+  class GetPreviewContent
+    def initialize(content_id:)
+      @content_id = content_id
+    end
+
+    def preview_content
+      {
+        title: content_item["title"],
+        html:,
+      }
+    end
+
+  private
+
+    def content_item
+      @content_item ||= begin
+        response = Services.publishing_api.get_content(@content_id)
+        response.parsed_content
+      end
+    end
+
+    def frontend_base_path
+      Rails.env.development? ? Plek.external_url_for("government-frontend") : Plek.website_root
+    end
+
+    def frontend_path
+      frontend_base_path + content_item["base_path"]
+    end
+
+    def html
+      uri = URI(frontend_path)
+      @html ||= Nokogiri::HTML.parse(Net::HTTP.get(uri))
```

Comment:
Could/should we have a fallback if we don't get a response? I'm worried that when we run locally, we'll have to have the frontend stack running too, which could get annoying

Reply:
hmm that's a good point and definitely something that needs some design thinking - what do you think we should show for now? A paragraph element that says "Preview not found"?

Reply:
Yeah, I think that'll be enough for now

Reply:
Yeah, I've just clocked that myself when testing - that's probably fine for now 👍